kyo-data/shared/src/main/scala/kyo/TypeMap.scala

```diff
@@ -12,11 +12,6 @@ opaque type TypeMap[+A] = TreeSeqMap[Tag[Any], Any]
 
 object TypeMap:
 
-    // TODO Tags of tags aren't supported yet. This marker trait reproduces the type-level
-    // behavior of `TypeMap` since `A` is covariant.
-    private trait TypeMapMarker[+A]
-    inline given [A]: Tag[TypeMap[A]] = Tag[TypeMapMarker[A]].asInstanceOf[Tag[TypeMap[A]]]
```

Comment:
I've changed the macro to not dealias opaque types so it doesn't even see `TreeSeqMap[Tag[Any], Any]` and doesn't attempt to obtain a `Tag[Tag[Any]]`